worlds/inscryption/__init__.py

```diff
@@ -0,0 +1,126 @@
+    location_name_to_id = {location: i + base_id for i, location in enumerate(all_locations)}
+
+    def create_item(self, name: str) -> Item:
+        item_id = self.item_name_to_id[name]
+        item_data = self.all_items[item_id - base_id]
+        return InscryptionItem(name, item_data['classification'], item_id, self.player)
+
+    def create_items(self) -> None:
+        nb_items_added = 0
+        useful_items = (act1_items + act2_items + act3_items) if self.options.goal != Goal.option_first_act \
+            else act1_items
+
+        if self.options.goal != Goal.option_first_act:
+            if self.options.epitaph_pieces_randomization.value == EpitaphPiecesRandomization.option_all_pieces:
+                useful_items.remove(act2_items[3])
+            elif self.options.epitaph_pieces_randomization.value == EpitaphPiecesRandomization.option_in_groups:
+                useful_items.remove(act2_items[2])
+                useful_items[len(act1_items) + 2]['count'] = 3
+            else:
+                useful_items.remove(act2_items[2])
+                useful_items[len(act1_items) + 2]['count'] = 1
+
+        for item in useful_items:
+            for _ in range(item["count"]):
+                new_item = self.create_item(item['name'])
+                self.multiworld.itempool.append(new_item)
+                nb_items_added += 1
+
+        filler_count = len(self.all_locations if self.options.goal != Goal.option_first_act else act1_locations)
+        filler_count -= nb_items_added
+
+        for i in range(filler_count):
+            index = i % len(filler_items)
+            filler_item = filler_items[index]
+            new_item = self.create_item(filler_item['name'])
+            self.multiworld.itempool.append(new_item)
+
+    def create_regions(self) -> None:
+        used_regions = inscryption_regions_all if self.options.goal != Goal.option_first_act \
+            else inscryption_regions_act_1
+        for region_name in used_regions.keys():
+            self.multiworld.regions.append(Region(region_name, self.player, self.multiworld))
+
+        for region_name, region_connections in used_regions.items():
+            region = self.multiworld.get_region(region_name, self.player)
+            region.add_exits(region_connections)
+            region.add_locations({
+                location: self.location_name_to_id[location] for location in regions_to_locations[region_name]
+            })
+
+    def set_rules(self) -> None:
+        Rules.InscryptionRules(self).set_all_rules()
+        if self.options.painting_checks_balancing.value == PaintingChecksBalancing.option_balanced:
+            Rules.InscryptionRules(self).set_painting_rules()
+        elif self.options.painting_checks_balancing.value == PaintingChecksBalancing.option_force_filler and \
+                sum(item.classification == ItemClassification.filler for item in self.multiworld.itempool) >= 2:
```

Comment:
I don't think you need be checking for this, just exclude the locations?

Reply:
What check specifically?

Reply:
Might be easier to explain this whole thing in the server since it's a few different things at once for me